pennylane/templates/embeddings/basis.py

```diff
@@ -72,9 +73,16 @@ def circuit(feature_vector):
 
     def __init__(self, features, wires, do_queue=True, id=None):
 
-        if isinstance(features, int):
-            bin_string = f"{features:b}".zfill(len(wires))
-            features = [1 if d == "1" else 0 for d in bin_string]
+        if isinstance(features, list):
+            features = qml.math.stack(features)
+
+        if qml.math.shape(features) == ():
+            if not qml.math.is_abstract(features) and features >= 2 ** len(wires):
+                raise ValueError(
+                    f"Features must be of length {len(wires)}, got features={features} which is >= {2 ** len(wires)}"
```

Comment:
That's a new error message if I am not mistaken but I did not find a corresponding test?

Reply:
Previously this was the same case as lines 95-97, and was tested in the `TestInputs` class. I had to split this off into two cases: one where `features` is an integer and the other where `features` is a vector, as these are fundamentally different. I kept the same test though.